### ISSUE TITLE
Text selection in the table's header and body cells

### ISSUE THREAD

**Issue by maxagin** [iterative/vscode-dvc#2260]
Table improvements proposal based on research and comparison of different products.

## Header

**Select text in header cells:**
Airtable NO, Google sheets NO, Wandb NO, Monday NO, VSCode YES(Experiments, Timestamp) the rest NO

👉 Recommending removing text selection in the header cells completely. 

## Body

**Select text in body cells:**
Airtable NO, Google sheets NO, Wandb YES, Monday NO, VSCode YES

**Can select text after double click on cell:**
Airtable YES, Google sheets YES, Monday YES

**Can select text using double click on cell:**
Airtable NO, Google sheets NO, Wandb YES, Monday YES, VSCode YES

👉 Recommending:

1. Remove text selection in body cells option
2. Can select text on double click or by highlighting only after the click  on the cell
3. The cursor is the default until the cell editing option is activated with a click on it
4. After clicking on a cell display the total value. If the value does not fit, when highlighting user can still select the entire number
5. When the cell is clicked we need to show the editing style (border inside the cell for example)
6. After the click on cell - hide the copy icon

Please share with me your thoughts about this proposal.

**Comment by maxagin** [comment]
> There must be a way to copy file names / section names. Full name and partial name.
Table header improvements #2256

@shcheklein, in the case of "There must be" we could follow the same idea as introduced for the Body cells. But this is only if there is a good reason for it.  Can you please explain why would user need to copy textual information from the header. Thanks

**Comment by shcheklein** [comment]
> Can you please explain why would user need to copy textual information from the header.

People always want to copy things in all places in the interface. To put into a text, to send a name in chat, to put into a report, to put into a browser, etc, etc. It's a general rule - if something is not easy to copy most likely it's a bug.

> Can select text on double click or by highlighting only after the click on the cell

I don't like this option since we are introducing an extra state to it. Kinda an active cell. It is complicated. Also, let's first fix things that are broken (headers), then do improvements ... also duplicate of the copy button discussion?

**Comment by shcheklein** [comment]
Let's split this please and deduplicate with other things.

**Comment by maxagin** [comment]
People are familiar with what they use on daily basis. Proving otherwise is very difficult.

---

>Can you please explain why would user need to copy textual information from the header?

>>People always want to copy things in all places in the interface. To put into a text, to send a name in chat, to put into a report, to put into a browser, etc, etc. It's a general rule - if something is not easy to copy most likely it's a bug.

Just in case you have not seen it:

>Select text in header cells:
Airtable NO, Google sheets NO, Wandb NO, Monday NO, VSCode YES(Experiments, Timestamp) the rest NO

👉 However, as I have mentioned already we can follow the same idea as introduced for the Body cells (see the description). Please let me know if this is the case.

---

>Can select text on double click or by highlighting only after the click on the cell

>>I don't like this option since we are introducing an extra state to it. Kinda an active cell. It is complicated. 

Just in case you have not seen it:

>Select text in body cells:
Airtable NO, Google sheets NO, Wandb YES, Monday NO, VSCode YES

>Can select text after double click on cell:
Airtable YES, Google sheets YES, Monday YES

---

👉 I would suggest reviewing the proposal again, please. Thank you @shcheklein !

**Comment by shcheklein** [comment]
> Google sheets - it's ONE symbol

> Airtable

<img width="424" alt="Screen Shot 2022-08-29 at 9 51 20 AM" src="https://user-images.githubusercontent.com/3659196/187252986-d221e381-266b-4520-89b4-a26879accf4b.png">

You can click an copy

> W&B: The mitigate this:

<img width="104" alt="Screen Shot 2022-08-29 at 10 16 45 AM" src="https://user-images.githubusercontent.com/3659196/187257838-31936aa6-ddcf-4c36-a313-d91847815b8e.png">

> Monday - the do:

<img width="146" alt="Screen Shot 2022-08-29 at 10 17 48 AM" src="https://user-images.githubusercontent.com/3659196/187258347-345a84d6-2142-4389-993a-7b0c7ea87f77.png">

So, in all cases (except Excel that is a clear edge case) - it's possible to copy information. 


------

> the same idea as introduced for the Body cells (see the description). Please let me know if this is the case.

yep, I don't the recent idea for body cell - we keep complicating and complicating things to my mind

> Just in case you have not seen it:

Same as above - each service has its own reasons. We need to understand why and what is done, fundamentals, strive for simplicity.

**Comment by shcheklein** [comment]
And on more thing - in our case header is way way more critical vs all the services above. In our case we have path to files or a nested option - it's super important to be able to copy the full part, relative path, just that name, etc. It's a severe UX bug for not allowing that.

**Comment by maxagin** [comment]
## V1


Table improvements proposal based on research and comparison of different products.

## Header

**Select text in header cells:**
Airtable NO, Google sheets NO, Wandb NO, Monday NO, VSCode YES(Experiments, Timestamp) the rest NO

👉 Recommending removing text selection in the header cells completely. 

## Body

**Select text in body cells:**
Airtable NO, Google sheets NO, Wandb YES, Monday NO, VSCode YES

**Can select text after double click on cell:**
Airtable YES, Google sheets YES, Monday YES

**Can select text using double click on cell:**
Airtable NO, Google sheets NO, Wandb YES, Monday YES, VSCode YES

👉 Recommending:

1. Remove text selection in body cells option
2. Can select text on double click or by highlighting only after the click  on the cell
3. The cursor is the default until the cell editing option is activated with a click on it
4. After clicking on a cell display the total value. If the value does not fit, when highlighting user can still select the entire number
5. When the cell is clicked we need to show the editing style (border inside the cell for example)
6. After the click on cell - hide the copy icon

Please share with me your thoughts about this proposal.

**Comment by maxagin** [comment]
@shcheklein, the proposal was reformulated and equipped with video examples. Please let me know WDYT? Thanks

**Comment by shcheklein** [comment]
Is it about cell values only now? (not header?) - let's rename the ticket then (we have a separate one for the header, right?)

Tbh, this logic with active cells feels too complicated, including implementation, I don't see enough value in it. What problem are we going to solve with it- to remove the button or you see other reasons/motivation?

**Comment by maxagin** [comment]
> Is it about cell values only now? (not header?) - let's rename the ticket then (we have a separate one for the header, right?)

Both header and body have the same behaviour:

- Click to edit cell
- Can select and copy the total value
- No need to have a copy button anymore

>Tbh, this logic with active cells feels too complicated

All the revised products do pretty much the same (except for the Wandb, which is far from been perfect UX, IMO)

> What problem are we going to solve with it

1. clear and consistent behaviour when working with information in cells (header, body)
2. "total value" consistent approach: short -> click -> full -> copy
3. no movement while working with the table (copy button does not appear on hover). Better readability

**Comment by maxagin** [comment]
@shcheklein I have not got any arguments from you about this subject. Any comments?

**Comment by shcheklein** [comment]
@maxagin it's not a priority right now, sorry. Let's get back to it when we have more progress on things that are more important at the moment.
